src/main/java/core/basesyntax/strategy/impl/ReturnOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.OperationException;
+
+public class ReturnOperation implements Operation {
+    private final StorageDao storageDao;
+
+    public ReturnOperation(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void proceed(FruitTransaction fruitTransaction) {
+        if (storageDao.get(fruitTransaction.getFruitName()) == null) {
+            throw new OperationException("Fruit " + fruitTransaction.getFruitName()
+                    + " doesn't exist");
+        }
+        Fruit fruit = storageDao.get(fruitTransaction.getFruitName());
+        fruit.setQuantity(fruit.getQuantity() + fruitTransaction.getQuantity());
```

Comment:
You are calling `storageDao.get(fruitTransaction.getFruitName())` twice - once for the check and once for getting the `Fruit` object. It's better to call it once, store the result in a variable, and then use that variable to check if the fruit exists and to update its quantity.